include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -130,6 +130,22 @@ __find_start_point(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_
     return _split_point_t<_Index>{*__res, __index_sum - *__res + 1};
 }
 
+template <typename _Rng1, typename _Rng2, typename _value_t_rng1 = oneapi::dpl::__internal::__value_t<_Rng1>,
+          typename _value_t_rng2 = oneapi::dpl::__internal::__value_t<_Rng2>>
+constexpr auto
+__can_use_ternary_op(int)
+    -> decltype(true ? std::declval<_value_t_rng1>() : std::declval<_value_t_rng2>(), std::true_type{})
+{
+    return {};
+}
+
+template <typename _T1, typename _T2>
+constexpr auto
+__can_use_ternary_op(...) -> std::false_type
+{
+    return {};
+}
+
```

Comment:
I suggest adding the following, then using it, to hide some more of the ugly details.

```
template <typename T1, typename T2>
constexpr static bool __can_use_ternary_op_v = decltype(__can_use_ternary_op<T1,T2>(0))::value;
```

Reply:
Done